### DISCUSSION TITLE
How to apply injection rate or flux to a domain instead of a boundary.

### DISCUSSION THREAD

**Discussion by steam-simulator** [idaholab/moose#23431] [Q&A Modules: General] [answered]
Hello,

Instead of applying an injection rate to a boundary/surface, I want to apply it to a small volume/domain within my model. Is this possible? How do I do it in MOOSE? I’m asking because, usually, BCs act on a surface/boundary and not a domain. Thank you.

**Comment by GiudGiud** [comment, marked as answer]
Hello

You ll want to use a BodyForce or CoupledForce kernel to do that.
In what MOOSE type is the injection rate defined? A Function? A variable?

Guillaume

**Reply by maxnezdyur** [reply]
Is this an example of a use case for FunctionDiracSource [https://mooseframework.inl.gov/source/dirackernels/FunctionDiracSource.html](https://mooseframework.inl.gov/source/dirackernels/FunctionDiracSource.html)?

**Reply by GiudGiud** [reply]
For a point source this would work yes

**Reply by steam-simulator** [reply]
Hello Guillaume,
Thanks, and sorry for the late reply. I think both BodyForce and CoupledForce will work. In my case, the flux, hence the force term, is a function and is not proportional to the concentration of a variable, so the BodyForce Kernel did work. 
@maxnezdyur, the link you provided is not working (:

**Reply by GiudGiud** [reply]
link is fixed now
